src/resource_prover.rs

```diff
@@ -6,41 +6,40 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use crate::ack_manager::ACK_TIMEOUT_SECS;
+use crate::ack_manager::ACK_TIMEOUT;
 use crate::action::Action;
 use crate::event::Event;
 use crate::id::PublicId;
 use crate::messages::{DirectMessage, MAX_PART_LEN};
 use crate::outbox::EventBox;
-use crate::signature_accumulator::ACCUMULATION_TIMEOUT_SECS;
+use crate::signature_accumulator::ACCUMULATION_TIMEOUT;
 use crate::state_machine::Transition;
+use crate::time::{Duration, Instant};
 use crate::timer::Timer;
 use crate::types::RoutingActionSender;
 use crate::utils::DisplayDuration;
-#[cfg(feature = "mock")]
-use fake_clock::FakeClock as Instant;
 use itertools::Itertools;
 use maidsafe_utilities::thread;
 use resource_proof::ResourceProof;
 use std::collections::HashMap;
 use std::iter::Iterator;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::Arc;
-use std::time::Duration;
-#[cfg(not(feature = "mock"))]
-use std::time::Instant;
 
 /// Time (in seconds) between accepting a new candidate (i.e. receiving an `AcceptAsCandidate` from
 /// our section) and sending a `CandidateApproval` for this candidate. If the candidate cannot
 /// satisfy the proof of resource challenge within this time, no `CandidateApproval` is sent.
-pub const RESOURCE_PROOF_DURATION_SECS: u64 = 300;
+pub const RESOURCE_PROOF_DURATION: Duration = Duration::from_secs(300);
 /// Maximum time a new node will wait to receive `NodeApproval` after receiving a
 /// `RelocateResponse`. This covers the built-in delay of the process and also allows time for the
 /// message to accumulate and be sent via four different routes.
-const APPROVAL_TIMEOUT_SECS: u64 =
-    RESOURCE_PROOF_DURATION_SECS + ACCUMULATION_TIMEOUT_SECS + (4 * ACK_TIMEOUT_SECS);
+const APPROVAL_TIMEOUT: Duration = Duration::from_secs(
+    RESOURCE_PROOF_DURATION.as_secs()
```

Comment:
Do these not work directly on the time? instead of back and forth conversion to secs?

Reply:
The `+` operator isn't `const` function, so can't be used in `const` definitions yet ([Rust playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a1ef3b3529ddb9298785e7b333b77389))